### DISCUSSION TITLE
Livewire not triggered when using over a network

### DISCUSSION THREAD

**Discussion by pkd2989** [livewire/livewire#2871] [1. Help]
I deployed my Laravel-Livewire project on apache server. The Livewire components were not working (they didn't trigger). After searching a bit, I exported the "livewire-assets" and made changes in the config\livewire.php
'asset_url' => 'http://localhost/aps/public',

Now it was working fine from the same computer, but when tried it from another system using the IP address of server, it was not working.

Then I changed the 'asset_url' to exact IP address instead of localhost. It started working.
Now the issue is, the IP address can change. How can get it working without hardcoding the IP in config file(and restarting the server).

Did anyone faced such issue. Please help.

**Comment by ju5t** [comment]
Apache has been discussed a few times in the past. It basically comes down to following the instruction from [Laravel](https://laravel.com/docs/7.x/installation#web-server-configuration). Also, you shouldn't make the root directory of Laravel public. It should be the `public` directory. If you do that, you shouldn't need to change anything in Livewire.

To be fair though, both aren't really Livewire related. It's a misconfiguration of Laravel on your server.

**Reply by pkd2989** [reply]
Hi, Please let me know how is that not a Livewire related. As my native laravel app works fine, without any need to meddle the config. Its just the livewire components that doesn't do its 'actions'.

**Reply by ju5t** [reply]
Because you have misconfigured Laravel, not Livewire. Your virtual host should point to the public folder, in your case the public folder is in the url. It shouldn't be there. As the link I posted earlier explains:

> Laravel should always be served out of the root of the "web directory" configured for your web server. You should not attempt to serve a Laravel application out of a subdirectory of the "web directory". Attempting to do so could expose sensitive files present within your application.

If you follow the instructions from Laravel you won't have any issues with Livewire either.
